<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Viever</class>
 <widget class="QMainWindow" name="Viever">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1134</width>
    <height>743</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Viever</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label_edges_amount">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>630</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>10</y>
      <width>401</width>
      <height>701</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="edges_amount">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>630</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Edges amount:</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_move_z">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>410</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-1000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_move_y">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>369</y>
       <width>67</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Move y:</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_scale">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>290</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="decimals">
      <number>2</number>
     </property>
     <property name="minimum">
      <double>0.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
     <property name="value">
      <double>1.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_open_result">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>321</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_vertexes">
     <property name="geometry">
      <rect>
       <x>21</x>
       <y>131</y>
       <width>60</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Vertixes:</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_rotate_z">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>530</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-360.000000000000000</double>
     </property>
     <property name="maximum">
      <double>360.000000000000000</double>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_open_choose_file">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>10</y>
       <width>131</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Choose obj file</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_figure_color">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>160</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton {
	background-color: rgb(53, 132, 228);
	color: black;
}</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="vertexes_amount">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>600</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_filename">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>570</y>
       <width>81</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Filename:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_z">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>530</y>
       <width>67</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotate z</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_rotate_y">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>490</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-360.000000000000000</double>
     </property>
     <property name="maximum">
      <double>360.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_vertexes_amount">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>600</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Vertexes amount:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_x">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>449</y>
       <width>67</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotate x:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_model_color">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>160</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Model color:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_move_x">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>329</y>
       <width>67</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Move x:</string>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>132</y>
       <width>207</width>
       <height>25</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QRadioButton" name="radioButton_none">
        <property name="text">
         <string>none</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QRadioButton" name="radioButton_circle">
        <property name="text">
         <string>circle</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QRadioButton" name="radioButton_square">
        <property name="text">
         <string>square</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_move_x">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>330</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-1000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_rotate_x">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>450</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-360.000000000000000</double>
     </property>
     <property name="maximum">
      <double>360.000000000000000</double>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_background_color">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>160</y>
       <width>41</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton {
	background-color: rgb(222, 221, 218)
	color: black;
}</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_exit">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>10</y>
       <width>121</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>EXIT</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_thickness">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>200</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Thickness:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_apply_changes">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>240</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>apply changes</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_projection">
     <property name="geometry">
      <rect>
       <x>21</x>
       <y>71</y>
       <width>74</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Projection:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_move_z">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>409</y>
       <width>67</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Move z:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_y">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>489</y>
       <width>67</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotate y:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_scale">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>290</y>
       <width>67</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Scale:</string>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>102</y>
       <width>141</width>
       <height>25</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QRadioButton" name="radioButton_solid">
        <property name="text">
         <string>solid</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QRadioButton" name="radioButton_dashed">
        <property name="text">
         <string>dashed</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="label_edges">
     <property name="geometry">
      <rect>
       <x>21</x>
       <y>101</y>
       <width>45</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Edges:</string>
     </property>
    </widget>
    <widget class="QLabel" name="filename">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>570</y>
       <width>211</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_background_color">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>160</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Background color:</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_move_y">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>370</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-1000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_clear_changes">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>240</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>clear changes</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_thikness">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>201</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="minimum">
      <double>1.000000000000000</double>
     </property>
     <property name="maximum">
      <double>100.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>1.000000000000000</double>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_bmp">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>670</y>
       <width>89</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>bmp</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_jpeg">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>670</y>
       <width>89</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>jpeg</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_gif">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>670</y>
       <width>89</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>gif</string>
     </property>
    </widget>
    <widget class="QLabel" name="edges_amount_2">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>630</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>70</y>
       <width>157</width>
       <height>25</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QRadioButton" name="radioButton_parallel">
        <property name="text">
         <string>parallel</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QRadioButton" name="radioButton_central">
        <property name="text">
         <string>central</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="glView" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>10</y>
      <width>700</width>
      <height>700</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1134</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>glView</class>
   <extends>QOpenGLWidget</extends>
   <header>glview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
